dbusmock/mockobject.py

```diff
@@ -686,9 +686,12 @@ def mock_method(self, interface: str, dbus_method: str, in_signature: str, *args
         # checks, except for the case of an empty signature
         if in_signature == '' and len(args) > 0:
             raise TypeError('Fewer items found in D-Bus signature than in Python arguments')
-        m = dbus.connection.MethodCallMessage('a.b', '/a', 'a.b', 'a')
-        m.append(signature=in_signature, *args)
-        args = m.get_args_list()
+        try:
+            m = dbus.connection.MethodCallMessage('a.b', '/a', 'a.b', 'a')
+            m.append(signature=in_signature, *args)
+            args = m.get_args_list()
+        except TypeError as e:
+            raise dbus.exceptions.DBusException('Invalid arguments', name='org.freedesktop.DBus.Error.InvalidArgs')
```

Comment:
Makes sense. Can you please cover this in test_api.py?

Reply:
Updated the test, also changed the signal emitting to match now.